core/src/secondary_api.rs

```diff
@@ -0,0 +1,92 @@
+use axum::{
+    body::{Body, Bytes},
+    extract::Request,
+    middleware::Next,
+    response::Response,
+};
+use http::StatusCode;
+use lazy_static::lazy_static;
+use reqwest::Client;
+use tracing::error;
+
+lazy_static! {
+    static ref SECONDARY_API_FORWARDING_ENABLED: bool =
+        std::env::var("SECONDARY_API_FORWARDING_ENABLED")
+            .map(|s| s == "true")
+            .unwrap_or(false);
+    static ref IS_SECONDARY: bool = std::env::var("IS_SECONDARY")
+        .map(|s| s == "true")
+        .unwrap_or(false);
+    static ref CORE_SECONDARY_API_URL: String =
+        std::env::var("CORE_SECONDARY_API").unwrap_or_default();
+}
+
+fn should_forward(req: &Request<Body>) -> bool {
+    if *SECONDARY_API_FORWARDING_ENABLED && !*IS_SECONDARY {
+        if CORE_SECONDARY_API_URL.is_empty() {
+            error!("CORE_SECONDARY_API is not set");
+        }
+        // Forward all requests for paths that contain "/tables" or "/query_database"
+        req.uri().path().contains("/tables") || req.uri().path().contains("/query_database")
+    } else {
+        false
+    }
+}
+
+pub async fn forward_middleware(req: Request<Body>, next: Next) -> Result<Response, StatusCode> {
+    if !should_forward(&req) {
+        return Ok(next.run(req).await);
+    }
+
+    let client = Client::new();
+    let (parts, body) = req.into_parts();
+    let body_bytes: Bytes = match axum::body::to_bytes(body, usize::MAX).await {
```

Comment:
I presume that given we never deal with files, this simple redirection should cover our needs , right?

Reply:
Yeah not sure if it'd work with files, but not super important since we don't use them with this endpoint